torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -837,6 +873,30 @@ def test_async_record_function_end_callbacks(self):
                 ),
             )
 
+    @dist_init
+    def test_async_record_function_cbs_jit_call(self):
+        _disable_profiling_for_testing()
+        if self.rank == 1:
+            with torch.autograd.profiler.profile() as pf:
+                key = _build_rpc_profiling_key(
+                    RPCExecMode.ASYNC,
+                    torch.jit._qualified_name(my_script_func),
+                    "worker1",
+                    "worker0",
+                )
+                with torch.autograd.profiler.record_function(key) as rf:
+                    fut = rpc.rpc_async(
+                        worker_name(0), my_script_func, args=(torch.tensor(1),)
+                    )
+                    # Intentionally calling record_function internals
+                    torch.ops.profiler._call_end_callbacks_on_jit_fut(rf.handle, fut)
+                fut.wait()
+            events = pf.function_events
+            rpc_event = get_function_event(
+                events, torch.jit._qualified_name(my_script_func)
+            )
+            self.assertTrue(torch.jit._qualified_name(my_script_func) in rpc_event.name)
```

Comment:
What are we testing here that isn't already covered by `test_profiler_with_script_async_rpc`?

Reply:
They're similar, but in that test, we ensure that we can correctly call `rf._call_end_callbacks_on_fut`, but in this test we directly test the functionality of `torch.ops.profiler._call_end_callbacks_on_jit_fut`.